### PR TITLE
feat: Add data CMS using strapi

### PR DESCRIPTION
**Pull request by talwinder50**
closes: #6 

Signed-off-by: talwinder.kaur <talwinder.kaur@securekey.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -38,3 +38,16 @@ issuer-rest:
 	@echo "Building issuer-rest"
 	@mkdir -p ./build/bin
 	@cd ${ISSUER_REST_PATH} && go build -o ../../build/bin/issuer-rest main.go
+strapi-build:
+	@echo "Building strapi demo"
+	@mkdir -p ./build/bin
+	@cd ${STRAPI_DEMO_PATH} && go build -o ../../build/bin/strapi-demo main.go
```
Comment (sandrask) [inline Makefile]: Where is strapi demo path set?

File: cmd/strapi-demo/createdemodata/start.go [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+		"status":       "graduated",
+		"totalcredits": "200",
+		"course":       "Bachelors'in Computing Science",
+	}
+
+	createOrFetchRecord(client, authToken, parameters.adminURL+adminURLEndpoint, "POST", studentRecord1)
+	createOrFetchRecord(client, authToken, parameters.adminURL+studentCardsEndpoint, "POST", studentRecord2)
+	createOrFetchRecord(client, authToken, parameters.adminURL+transcriptEndpoint, "POST", transcriptRecord1)
+	createOrFetchRecord(client, authToken, parameters.adminURL+transcriptEndpoint, "POST", transcriptRecord2)
+	createOrFetchRecord(client, authToken, parameters.adminURL+studentCardsEndpoint, "GET", nil)
```
Comment (sandrask) [inline cmd/strapi-demo/createdemodata/start.go]: This get doesn't verify what was inserted, it merely lists all records. Can you add fetch by some id and verify that it equals inserted record?

File: Makefile (hunk 1 above)
Comment (talwinder50) [reply Makefile]: done

File: cmd/strapi-demo/createdemodata/start.go [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+	// to verify the record are successfully inserted
+	if method == "GET" {
+		var resp map[string]interface{}
+
+		err = json.Unmarshal(body, &resp)
+		if err != nil {
+			fmt.Println(err)
+		}
+
+		if resp["studentid"] == studentRecord1["studentid"] || resp["studentid"] == transcriptRecord1["studentid"] {
```
Comment (sandrask) [inline cmd/strapi-demo/createdemodata/start.go]: Verification should not be here

File: cmd/strapi-demo/createdemodata/start.go [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+	authToken, err := createAdminUser(client, parameters.adminURL, adminUserValues)
+	if err != nil {
+		return err
+	}
+
+	createOrFetchRecord(client, authToken, parameters.adminURL+studentCardsEndpoint, "POST", studentRecord1)
+	createOrFetchRecord(client, authToken, parameters.adminURL+studentCardsEndpoint, "POST", studentRecord2)
+	createOrFetchRecord(client, authToken, parameters.adminURL+transcriptEndpoint, "POST", transcriptRecord1)
+	createOrFetchRecord(client, authToken, parameters.adminURL+transcriptEndpoint, "POST", transcriptRecord2)
+	createOrFetchRecord(client, authToken, parameters.adminURL+studentCardsEndpoint+"/1", "GET", studentRecord1)
```
Comment (sandrask) [inline cmd/strapi-demo/createdemodata/start.go]: createOrFetchRecord function should return response and then you should verify that response against studentRecord1. You should not add verification to createOrFetchRecord. 

This way you don't need global variables so you can get rid of nolint: gochecknoglobals. 

If you want to keep same interface for your function you can do something like:
createOrFetchRecord(client, authToken, parameters.adminURL+studentCardsEndpoint, "POST", studentRecord1)
var cmsSudentRecord1 map[string]interface{}
createOrFetchRecord(client, authToken, parameters.adminURL+studentCardsEndpoint+"/1", "GET", cmsStudentRecord1)
check if cmsStudentRecord1 data matches studentRecord1

**Pull request merged**